module/reset/include/mod_reset.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Arm SCP/MCP Software
+ * Copyright (c) 2015-2019, Arm Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Description:
+ *     Power domain management support.
+ */
+
+#ifndef MOD_RESET_DOMAIN_H
+#define MOD_RESET_DOMAIN_H
+
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <fwk_element.h>
+#include <fwk_id.h>
+#include <fwk_module_idx.h>
+
+/*!
+ * \addtogroup GroupModules Modules
+ * @{
+ */
+
+/*!
+ * \defgroup GroupResetDomain Reset Domains
+ *
+ * \details Support for setting the state of reset domains.
+ *
+ * @{
+ */
+
+/*!
+ * \brief APIs that the module makes available to entities requesting binding.
+ */
+enum mod_reset_api_type {
+    /*! Reset HAL */
+    MOD_RESET_API_TYPE_HAL,
+    MOD_RESET_API_COUNT,
+};
+
+
+/*!
+ * \brief Reset module configuration data.
+ */
+struct mod_reset_config {
+    /*!
+     * \brief Identifier of a notification to subscribe reset domain devices to
+     *     in order to receive notifications of domain reset that have already
+     *     occurred.
+     *
+     * \note May be \ref FWK_ID_NONE to disable this functionality for all
+     *     elements.
+     *
+     *  TODO: this is not yet implemented in the driver.
+     */
+    const fwk_id_t notification_id;
+};
+
+/*!
+ * \brief Reset domain element configuration data.
+ */
+struct mod_reset_dev_config {
+    /*! Reference to the device element within the associated driver module */
+    const fwk_id_t driver_id;
```

Comment:
Since I see drivers being referenced, is this a HAL header file then? Could you please insert an appropriate comment.

Reply:
Sorry, I don't understand. What do you mean by HAL header file?

Reply:
I meant to ask if this is a HAL or a driver related header?
The convention we use is SCMI->HAL->Driver.

Reply:
Ok, then mod_reset a HAL.